packages/blaze-data/src/index.ts

```diff
@@ -505,21 +515,29 @@ export function _parse<T extends TSchema>(
       );
     }
     const actualCtor = value.getAlternative();
-    const variantIndex = type.anyOf.findIndex((s) => {
-      const resolved = resolveType(s, path, defs);
-      return "ctor" in resolved && BigInt(resolved["ctor"]) === actualCtor;
-    });
-    if (variantIndex < 0) {
+
+    let variant: TSchema | undefined;
+    let variantName: string | undefined;
+    for (let idx = 0; idx < type.anyOf.length; idx++) {
+      const v = type.anyOf[idx]!;
+      let resolved = resolveType(v, path, defs);
+      let key = idx.toString();
+      if ("properties" in resolved) {
+        const properties = resolved["properties"];
+        key = Object.keys(properties)[0]!;
+        resolved = properties[key];
+      }
```

Comment:
What case is this solving for? If a union type has a field named `properties`, does it definitely always have a single property which is the _actual_ type?

Reply:
Union variants with no fields turn into strings, with the ctor property on the direct child of the anyOf.

But if the variant has fields, it maps to an object with the variant name as the key, and the object is one level down.

So for example, 
```
enum MultisigScript {
  Signature { key_hash: String }
  ...
}
```

Maps to a JavaScript type like:
```
{
  Signature: {
    key_hash: "..."
  }
}
`

Which means that we'll have a union, and each variant may be a string, or an object. If it's an object, the ctor is defined on the nested object, not the top level object.